File: BCC/EDI (CCB0714)/python/estacionamento/model/estacionamento.py
from fila.model.fila import Fila

class Estacionamento():
    def __init__(self, tamanhoPrincipal=10):
        self.principal = Fila(); # Estacionamento principal
        self.espera = Fila() # Estacionamento de espera
        self.tamanhoPrincipal = tamanhoPrincipal

    def colocaCarro(self, placa):
        if (self.principal.size() < self.tamanhoPrincipal):
            self.principal.insereNaFila(placa)
        else:
            self.espera.insereNaFila(placa)

    def retiraCarro(self):
        if (self.principal.size() >= self.tamanhoPrincipal):
            if (self.espera.isEmpty is not False):
                _retiradoEspera = self.__retiraCarroEspera()
                self.principal.insereNaFila(_retiradoEspera)
            return self.principal.removeDaFila()
        else:
            return self.principal.removeDaFila()

    def __retiraCarroEspera(self):
        return self.espera.removeDaFila()
    
    def listaPrincipal(self):
        self.principal.mostraFila()

    def listaEspera(self):
        self.espera.mostraFila()

File: BCC/EDI(CCB0714)/C/estacionamento/model/estacionamento.c
#include <stdio.h>
#include <stdlib.h>
#include "../../fila/model/fila.c"

//Funções do estacionamento
void estacionar(Lista*, Lista*, int);
int partida(Lista*, Lista*);

//Implementação
void estacionar(Lista* principal, Lista* espera, int placa){
    if (tamanhoFila(principal) < 10){
        insereNaFila(principal, placa);
    } else {
        insereNaFila(espera, placa);
    }
}

int partida(Lista* principal, Lista* espera){
    int placaRemovida, placa;
    if ((tamanhoFila(principal) > 9) && (filaVazia(espera) != 1)){
        placaRemovida = removeDaFila(principal);
        placa = removeDaFila(espera);
        insereNaFila(principal, placa);
    } else {
        placaRemovida = removeDaFila(principal);
    }
    return placaRemovida;
}

